### PR TITLE
Replace some deprecated method calls in guava

### PR DESCRIPTION
**Pull request by navis**
Replaced Input/OutputSupplier with ByteSource/Sink

### CODE REVIEW

File: common/src/main/java/io/druid/common/guava/FileOutputSupplier.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 
 /**
 */
+@Deprecated
```
Comment (drcrallen) [inline common/src/main/java/io/druid/common/guava/FileOutputSupplier.java]: Needs a "use XXX instead" note

File: common/src/main/java/io/druid/common/guava/GuavaUtils.java [hunk 2]
```diff
@@ -52,47 +51,36 @@ public String apply(@Nullable String input)
...
+              public CharSource apply(final File input)
+              {
+                return new CharSource()
+                {
+                  @Override
+                  public Reader openStream() throws IOException
+                  {
+                    InputStream baseStream = new FileInputStream(input);
+                    if (input.getName().endsWith(".gz")) {
+                      baseStream = new GZIPInputStream(baseStream);
```
Comment (drcrallen) [inline common/src/main/java/io/druid/common/guava/GuavaUtils.java]: While we're here can you make this wrapped by `com.metamx.common.CompressionUtils#gzipInputStream` ?

**Comment by xvrl** [comment]
@navis this is a tricky one, there are lots of thorny Guava issues when running on hadoop, which ships an old version of guava. This means we cannot use anything beyond what is available in Guava 11 for any code that could be executed during hadoop indexing.

File: processing/src/main/java/io/druid/segment/MetricHolder.java [hunk 3]
```diff
@@ -109,6 +106,9 @@ public static void writeToChannel(MetricHolder holder, WritableByteChannel out)
     serializerUtils.writeString(out, holder.typeName);
 
     switch (holder.type) {
+      case LONG:
```
Comment (drcrallen) [inline processing/src/main/java/io/druid/segment/MetricHolder.java]: Was this added because the enum was missing?

**Comment by drcrallen** [comment]
@xvrl is correct, do you happen to have the guava version these things were added in?

File: processing/src/main/java/io/druid/segment/MetricHolder.java (hunk 3 above)
Comment (navis) [inline processing/src/main/java/io/druid/segment/MetricHolder.java]: Yes, looks like missed by mistake.

**Comment by navis** [comment]
Started this after seeing javadoc of OutputSupplier that it's `scheduled for removal in June 2015`. But as @xvrl said, I've apparently overlooked hadoop compatibility issue. It would be better to postpone this for later.

File: processing/src/main/java/io/druid/segment/MetricHolder.java (hunk 3 above)
Comment (drcrallen) [inline processing/src/main/java/io/druid/segment/MetricHolder.java]: This would still be worth bringing up in a separate PR, and bonus points if you can spot the unit test which missed it.

**Comment by fjy** [comment]
@navis is it okay if I close this for now? Please reopen when you think it is time to look at these changes again

**Comment by navis** [comment]
Closing, for now.
